core/ulog/builder_test.go

```diff
@@ -0,0 +1,144 @@
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package ulog
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"go.uber.org/fx/core/config"
+	"go.uber.org/fx/core/testutils"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/uber-go/zap"
+)
+
+var _environment = "_ENVIRONMENT"
+
+func TestConfiguredLogger(t *testing.T) {
+	withLogger(t, func(builder *LogBuilder, tmpDir string, logFile string) {
+		txt := false
+		cfg := Configuration{
+			Level:         "debug",
+			Stdout:        false,
+			TextFormatter: &txt,
+			Verbose:       false,
+		}
+		log := builder.WithConfiguration(cfg).Build()
+		zaplogger := log.RawLogger()
+		assert.Equal(t, zap.DebugLevel, zaplogger.Level())
+	})
+}
+
+func TestConfiguredLoggerWithTextFormatter(t *testing.T) {
+	withLogger(t, func(builder *LogBuilder, tmpDir string, logFile string) {
+		txt := true
+		cfg := Configuration{
+			Level:         "debug",
+			Stdout:        false,
+			TextFormatter: &txt,
+			Verbose:       false,
+			File: &FileConfiguration{
+				Directory: tmpDir,
+				FileName:  logFile,
+				Enabled:   true,
+			},
+		}
+		log := Builder().WithConfiguration(cfg).Build()
+		zaplogger := log.RawLogger()
+		assert.Equal(t, zap.DebugLevel, zaplogger.Level())
+	})
+}
+
+func TestConfiguredLoggerWithTextFormatter_NonDev(t *testing.T) {
+	withLogger(t, func(builder *LogBuilder, tmpDir string, logFile string) {
+		txt := true
+		log := Builder().WithConfiguration(Configuration{
+			Level:         "debug",
+			TextFormatter: &txt,
+		}).Build()
+		zaplogger := log.RawLogger()
+		assert.Equal(t, zap.DebugLevel, zaplogger.Level())
+	})
+}
+
+func TestConfiguredLoggerWithStdout(t *testing.T) {
+	withLogger(t, func(builder *LogBuilder, tmpDir string, logFile string) {
+		txt := false
+		cfg := Configuration{
+			Stdout:        true,
+			TextFormatter: &txt,
+			Verbose:       true,
+			File: &FileConfiguration{
+				Enabled:   true,
+				Directory: tmpDir,
+				FileName:  logFile,
+			},
+		}
+		log := Builder().WithConfiguration(cfg).Build()
+		zaplogger := log.RawLogger()
+		assert.Equal(t, zap.DebugLevel, zaplogger.Level())
+	})
+}
+
+func withLogger(t *testing.T, f func(*LogBuilder, string, string)) {
+	defer testutils.EnvOverride(t, config.GetEnvironmentPrefix()+_environment, "madeup_env")()
```

Comment:
`config.GetEnvironmentPrefix()+_environment, "madeup_env"` is ugly

lets make a `config.Env(env stirng)` function that auto-prefixes